### PR TITLE
build: refactor build process

### PR DESCRIPTION
**Pull request by xballoy**
⚠️ **Work in progress** ⚠️

### TL;DR;

- [x] Use Babel to compile all files and tsc only to generate declaration files
- [x] Add components' endpoint in `react-toolkit-all` to allow tree shaking
- [ ] Remove `style.js` and allow each component to build its own style

🚨 Current issues:
- [ ] Unable to run jest tests on JavaScript files: `babel-jest: Babel ignores packages/Layout/header/src/NavBar/Navbar.helpers.spec.js - make sure to include the file in Jest's transformIgnorePatterns as well.`

### Use babel to compile all files and tsc only to generate declaration files

- Babel is faster to transpile TypeScripts files because it doesn't do type checking.
- We use the same pipeline to compile all files

### Add components' endpoint in `react-toolkit-all` to allow tree shaking

Add the following missing packages in the `index.js` of the `react-toolkit-all` package:

- @axa-fr/react-toolkit-form-filter
- @axa-fr/react-toolkit-form-filter-inline
- @axa-fr/react-toolkit-highlight
- @axa-fr/react-toolkit-link
- @axa-fr/react-toolkit-list
- @axa-fr/react-toolkit-panel
- @axa-fr/react-toolkit-status
- @axa-fr/react-toolkit-title

There was a duplicate export so I had rename Title to HeaderTitle in `react-toolkit-layout-header` because it was a duplicate export with Title in title.
Maybe that `react-toolkit-title` wasn't in `react-toolkit-all` due to that issue ¯\_(ツ)_/¯

Because it feels weird to use to do the following in an application
```
import { Button } from '@axa-fr/react-toolkit-all/dist/button'
```
I renamed the `dist` folder to `component` for `react-toolkit-all` (not the other components) so we can do the following
```
import { Button } from '@axa-fr/react-toolkit-all/component/button'
```

With those new endpoints we have a huge bundle size improvement:

| Case                                                                                              | Size (KB) |
| ------------------------------------------------------------- | --------- |
| Empty CRA project                                                                        | 134,63    |
| Adding a button (using @axa-fr/react-toolkit-all)                       | 1040      |
| Adding a button (using @axa-fr/react-toolkit-all/component/button)   | 155,41    |
| Adding a button (using @axa-fr/react-toolkit-button)                | 155,21    |

### Remove `style.js` and allow each component to build its own style

- Unable to "watch" style
- Building the style is a separate task

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ tsconfig.json
 tsconfig.dev.json
 tsconfig.test.json
 package.json
+packages/all/*.js
```
Comment (xballoy) [inline .eslintignore:15]: Can be removed

File: .prettierignore [hunk 2]
```diff
@@ -8,3 +8,4 @@ CHANGELOG.md
 lerna.json
 packages/all/src/bootstrap/af-components.template.scss
 packages/all/src/bootstrap/af-toolkit-core.template.scss
+packages/all/*.js
```
Comment (xballoy) [inline .prettierignore:11]: Can be removed

File: packages/all/package.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@axa-fr/react-toolkit-all",
   "version": "1.4.0-alpha.0",
-  "main": "dist/index.js",
+  "main": "index.js",
```
Comment (xballoy) [inline packages/all/package.json:4]: Change to component/index.js

File: packages/all/src/layout-header.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export {
+  Header,
+  Infos,
+  Name,
+  NavBar,
+  NavBarBase,
+  NavBarItem,
+  NavBarItemBase,
+  NavBarItemLink,
+  Title as HeaderTitle,
```
Comment (xballoy) [inline packages/all/src/layout-header.js:10]: This is a breaking change, add it to the migration guide

File: packages/all/package.json [hunk 5]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@axa-fr/react-toolkit-all",
   "version": "1.4.0-alpha.0",
-  "main": "dist/index.js",
+  "main": "component/index.js",
```
Comment (xballoy) [inline packages/all/package.json:4]: Add some documentation

**Review by arnaudforaison — APPROVED** [review]

**Comment by xballoy** [comment]
@arnaudforaison tu me fais signe quand vous aurez défini une gestion des breaking change, je fixerais les conflits et mettrais la PR à jour

**Comment by arnaudforaison** [comment]
C'est fait. Tu peux mettre à jour et rien ne change on merge sur master, breaking changes inclus.
